### PR TITLE
Structure for the flood_forecaster client tool

### PR DESCRIPTION
**Pull request by acm7**
## Description of changes
* add structure for the common client tool
* refactor schema_report and schema_update scripts as commands of `flood_forecaster_cli` tool 
* restructure unittests; add unittests for database_helper
* add `fetch_table_to_csv` method in `database_helper` to dump a schema.table from PostgreSQL

## How did we test it
- run unittests `pytest -s -v src/tests/`
- run client tool `flood_forecaster_cli database-model list-db-schemas` 

## Any relevant logs or outputs
```
$ pytest -s -v src/tests/ 
======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.9.5, pytest-7.3.1, pluggy-1.0.0 -- /home/amihailescu/dev/myenv/py39/bin/python
cachedir: .pytest_cache
rootdir: /home/amihailescu/dev/github/saadaal-flood-forecaster
plugins: cov-4.0.0, anyio-4.2.0
collected 7 items                                                                                                                                                                                    

src/tests/test_configuration.py::TestConfig::test_load_config_file_not_found PASSED
src/tests/test_configuration.py::TestConfig::test_load_config_success PASSED
src/tests/test_configuration.py::TestConfig::test_load_database_config PASSED
src/tests/test_configuration.py::TestConfig::test_load_openmeteo_config PASSED
src/tests/test_database_helper.py::TestDatabaseHelper::test_database_connection_failure PASSED
src/tests/test_database_helper.py::TestDatabaseHelper::test_database_connection_success Connected to database 'testdb'
PASSED
src/tests/test_database_helper.py::TestDatabaseHelper::test_missing_env_password PASSED
```

```
flood_forecaster_cli database-model list-db-schemas
Connected to database 'postgres'
Available schemas: ['flood_forecaster', 'hdb_catalog', 'information_schema', 'public']
Schemas in the database:
- flood_forecaster
- hdb_catalog
- information_schema
- public
```

### CODE REVIEW

File: src/flood_forecaster_cli/commands/data_ingestion.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Commands for data ingestion
+    """
+
+
+@data_ingestion.command("load-csv", help="Load data from csv file to db schema.table")
+@click.option("--file_path", "-f", required=True, help="Path to file")
+@click.option('--schema', '-s', required=True, help='Target database schema.')
+@click.option('--table', '-t', required=True, help='Target database table.')
+@common_options
+def load_csv(configuration: Config, file_path: str, schema_name: str, table_name: str):
```
Comment (FlorEtcheverry) [inline src/flood_forecaster_cli/commands/data_ingestion.py:26]: Add in the CLI a disclaimer saying it's not yet implemented? (in the help maybe?)

File: src/flood_forecaster_cli/commands/database_model.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+"""
+Data modelling Commands
+"""
+
+import click
+from flood_forecaster.utils.configuration import Config
+from flood_forecaster.utils.database_helper import DatabaseConnection
+# from flood_forecaster.utils.logging import logger
```
Comment (FlorEtcheverry) [inline src/flood_forecaster_cli/commands/database_model.py:8]: Do we have a logger? It was in my "improvements" list, I think

File: src/tests/test_database_helper.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import sys
+import os
+
+import unittest
+from unittest.mock import patch, MagicMock
+from flood_forecaster.utils.database_helper import DatabaseConnection
+# from flood_forecaster.utils.configuration import Config
+# from sqlalchemy.exc import SQLAlchemyError
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../")))
```
Comment (FlorEtcheverry) [inline src/tests/test_database_helper.py:10]: What is this for?

File: src/flood_forecaster/utils/database_helper.py [hunk 4]
```diff
@@ -22,7 +23,7 @@ def __init__(self, config: Config, db_password: str = None) -> None:
         self.user = config.get("user")
         self.host = config.get("host")
         self.port = int(config.get("port"))
-        self.password = self._get_env_pwd() if db_password is None else db_password
+        self.password = os.environ.get("POSTGRES_PASSWORD") if db_password is None else db_password
```
[the remark this answers is not in the document]
Comment (acm7) [reply src/flood_forecaster/utils/database_helper.py:26]: Actually it was always raising an err - as no pwd was returned , so did the fallback; will commit the fix quickly.

File: src/flood_forecaster_cli/commands/data_ingestion.py (hunk 1 above)
Comment (acm7) [reply src/flood_forecaster_cli/commands/data_ingestion.py:26]: Comment is in the code already; sample command was added as place holder, but could be removed if not used.

File: src/flood_forecaster_cli/commands/database_model.py (hunk 2 above)
Comment (acm7) [reply src/flood_forecaster_cli/commands/database_model.py:8]: Not yet - in the todo list, yet not a priority.

File: src/tests/test_database_helper.py (hunk 3 above)
Comment (acm7) [reply src/tests/test_database_helper.py]: leftover removed now

**Review by FlorEtcheverry — APPROVED** [review]

**Pull request merged**